pkg/ciliumenvoyconfig/cec_reconciler_test.go

```diff
@@ -0,0 +1,673 @@
+				{Namespace: "ns1", Name: "config1"}: testConfig("ns1", "config1", nil, true),
+			},
+			currentNodeLabels: map[string]string{
+				"role": "infra",
+			},
+			newNodeLabels: map[string]string{
+				"role": "infra",
+			},
+			expectedError:   false,
+			expectedAdded:   []string{},
+			expectedDeleted: []string{},
+		},
+		{
+			name: "No changes if there are no changes in configs selecting nodes or not",
+			configs: map[resource.Key]*config{
+				{Namespace: "ns1", Name: "config1"}: testConfig("ns1", "config1", nil, true),
+				{Namespace: "ns1", Name: "config2"}: testConfig("ns1", "config2", nil, true),
+			},
+			currentNodeLabels: map[string]string{
+				"role": "infra",
+			},
+			newNodeLabels: map[string]string{
+				"role": "worker",
+			},
+			expectedError:   false,
+			expectedAdded:   []string{},
+			expectedDeleted: []string{},
+		},
+		{
+			name: "Delete configs that no longer select the local node",
+			configs: map[resource.Key]*config{
+				{Namespace: "ns1", Name: "config1"}: testConfig("ns1", "config1", map[string]string{"role": "infra"}, true),
+				{Namespace: "ns1", Name: "config2"}: testConfig("ns1", "config2", nil, true),
+			},
+			currentNodeLabels: map[string]string{
+				"role": "infra",
+			},
+			newNodeLabels: map[string]string{
+				"role": "worker",
+			},
+			expectedError:   false,
+			expectedAdded:   []string{},
+			expectedDeleted: []string{"ns1/config1"},
+		},
+		{
+			name: "Add configs that start to select the local node",
+			configs: map[resource.Key]*config{
+				{Namespace: "ns1", Name: "config1"}: testConfig("ns1", "config1", map[string]string{"role": "infra"}, false),
+				{Namespace: "ns1", Name: "config2"}: testConfig("ns1", "config2", nil, true),
+			},
+			currentNodeLabels: map[string]string{
+				"role": "worker",
+			},
+			newNodeLabels: map[string]string{
+				"role": "infra",
+			},
+			expectedError:   false,
+			expectedAdded:   []string{"ns1/config1"},
+			expectedDeleted: []string{},
+		},
+		{
+			name: "Multiple changes",
+			configs: map[resource.Key]*config{
+				{Namespace: "ns1", Name: "config1"}: testConfig("ns1", "config1", map[string]string{"role": "infra", "node": "node1"}, false),
+				{Namespace: "ns1", Name: "config2"}: testConfig("ns1", "config2", map[string]string{"role": "infra", "node": "node1"}, false),
+				{Namespace: "ns1", Name: "config3"}: testConfig("ns1", "config3", map[string]string{"role": "infra", "node": "node1", "environment": "test"}, false),
+				{Namespace: "ns1", Name: "config4"}: testConfig("ns1", "config4", nil, true),
+				{Namespace: "ns1", Name: "config5"}: testConfig("ns1", "config5", map[string]string{"role": "worker", "node": "node1"}, true),
+				{Namespace: "ns1", Name: "config6"}: testConfig("ns1", "config6", map[string]string{"role": "worker", "node": "node1"}, true),
+				{Namespace: "ns1", Name: "config7"}: testConfig("ns1", "config7", map[string]string{"role": "worker", "node": "node1", "environment": "test"}, true),
```

Comment:
Looks like this should have been `false` as current node labels do not specify `environment`: `test`?

Reply:
as mentioned in the comment on the property `selectsLocalNode` of `config`: it's the pre-calculated boolean of this property that gets used when comparing whether an existing config has previously been applied to the xDS of the local node. Hence the logic is correct :crossed_fingers: 

But i agree that it should match with the labels in the testcase. Therefore i changed it accordingly to `false`.